### PR TITLE
Deploy chaos server by Helm 

### PR DESCRIPTION
**Pull request by cwen0**
### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,4 +30,5 @@ test/image/e2e/manifests/
 
 /pkg/uiserver/embedded_assets_handler.go
 
-/doc/swagger.*
+/docs/swagger.*
+/docs/docs.go
```
Comment (yeya24) [inline .gitignore:34]: 👍

File: helm/chaos-mesh/templates/chaos-server-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+{{ toYaml .Values.chaosServer.resources | indent 12 }}
+          command:
+            - /usr/local/bin/chaos-server
+          envFrom:
+          - configMapRef:
+            name: {{ template "chaos-mesh.name" . }}-chaos-server
+          volumeMounts:
+          - name: storage-volume
+            mountPath: {{ .Values.chaosServer.persistentVolume.mountPath }}
+            subPath: "{{ .Values.chaosServer.persistentVolume.subPath }}"
```
Comment (yeya24) [inline helm/chaos-mesh/templates/chaos-server-deployment.yaml:48]: Let's provide a set up with MySQL instead of sqlite?

**Comment by yeya24** [comment]
Do we decide to call this component `chaos-server` finally? Or `chaos-dashboard`?

**Comment by cwen0** [comment]
> Do we decide to call this component `chaos-server` finally? Or `chaos-dashboard`?

I think chaos-server is better. chaos-dashboard just is the UI of the chaos-server.

**Comment by yeya24** [comment]
> > Do we decide to call this component `chaos-server` finally? Or `chaos-dashboard`?
> 
> I think chaos-server is better. chaos-dashboard just is the UI of the chaos-server.

I think dashboard means both frontend and backend. Like `tidb-dashboard` or `kubernetes-dashboard`.

**Comment by cwen0** [comment]
> > > Do we decide to call this component `chaos-server` finally? Or `chaos-dashboard`?
> > 
> > 
> > I think chaos-server is better. chaos-dashboard just is the UI of the chaos-server.
> 
> I think dashboard means both frontend and backend. Like `tidb-dashboard` or `kubernetes-dashboard`.

At present we all use them as chaos-server, maybe we can rename them in other single PR.

File: helm/chaos-mesh/templates/chaos-server-deployment.yaml (hunk 2 above)
Comment (cwen0) [reply helm/chaos-mesh/templates/chaos-server-deployment.yaml:48]: Using sqlite3 directly can speed up our current test and simple the steps of deployment, and we also allow to set MySQL DNS for chaos server store using the ENV in `value.yaml` file.

Comment (yeya24) [reply helm/chaos-mesh/templates/chaos-server-deployment.yaml:48]: Great

File: helm/chaos-mesh/templates/chaos-server-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: chaos-server
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  selector:
+    app.kubernetes.io/name: {{ template "chaos-mesh.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: chaos-server
+  type: NodePort
```
Comment (yeya24) [inline helm/chaos-mesh/templates/chaos-server-deployment.yaml:87]: Does it need to have `NodePort` by default?

File: helm/chaos-mesh/values.yaml [hunk 4]
```diff
@@ -83,12 +83,12 @@ chaosDaemon:
 
   tolerations: []
 
-dashboard:
+chaosServer:
   create: false
 
-  serviceAccount: chaos-dashboard
+  serviceAccount: chaos-controller-manager
```
Comment (yeya24) [inline helm/chaos-mesh/values.yaml:89]: We use the same SA as controller-manager?

File: helm/chaos-mesh/templates/chaos-server-deployment.yaml (hunk 3 above)
Comment (cwen0) [reply helm/chaos-mesh/templates/chaos-server-deployment.yaml:87]: I think it is ok. We need to use it in the IDC environment.

Comment (yeya24) [reply helm/chaos-mesh/templates/chaos-server-deployment.yaml:87]: OK.  But we can change this later or make this service type configurable

File: helm/chaos-mesh/values.yaml (hunk 4 above)
Comment (cwen0) [reply helm/chaos-mesh/values.yaml:89]: Most parts are the same, we use the same one at present and I will pull the other pr to add a new serviceAccount for Chaos Server.

**Pull request merged**